contentcuration/contentcuration/utils/publish.py

```diff
@@ -226,14 +227,19 @@ def create_bare_contentnode(ccnode, default_language, channel_id, channel_name):
     if ccnode.language or default_language:
         language, _new = get_or_create_language(ccnode.language or default_language)
 
+    options = {}
+    if ccnode.extra_fields and 'options' in ccnode.extra_fields:
+        options = ccnode.extra_fields['options']
+
     duration = None
     if ccnode.kind_id in [content_kinds.AUDIO, content_kinds.VIDEO]:
-        # aggregate duration from associated files, choosing maximum if there are multiple, like hi and lo res videos
+        # aggregate duration from associated files, choosing maximum if there are multiple, like hi and lo res videos.
         duration = ccnode.files.aggregate(duration=Max("duration")).get("duration")
 
-    options = {}
-    if ccnode.extra_fields and 'options' in ccnode.extra_fields:
-        options = ccnode.extra_fields['options']
+    ccnode_completion_criteria = options.get("completion_criteria")
+    if ccnode_completion_criteria:
```

Comment:
There's a chance for optimizing the order of these operations here. Any SQL queries will be slower than staying within Python, so it would likely be beneficial for the performance of publishing to avoid making the query when we know we'll discard the result.

Reply:
Ohh, really nice catch there!